### PR TITLE
Bug/44 fix connection

### LINKED ISSUE #44: Fix SSH connection
Connection fails on windows
Additionally error message should be improved to show actual data for 
* user
* host
* port

Old implementation:
```java 
throw new IllegalStateException(ExaError.messageBuilder("E-ETAJ-12")
  .message("Failed to ssh to the exasol database. This is required for redirecting a host port.")
  .mitigation("Make sure the database is reachable (port 22 open)")
  .mitigation("Make sure 'ec2-user' can login to the database using your ssh-key.").toString(),
     exception);
```

### PR DESCRIPTION
**Pull request by ckunki**
fixes #44

### CODE REVIEW

File: src/main/java/com/exasol/exasoltestsetup/SessionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+import java.util.Hashtable;
+
+import com.jcraft.jsch.*;
+
+/**
+ * Builder for {@link com.jcraft.jsch.Session}
+ */
+public class SessionBuilder {
+    // session parameters
```
Comment (morazow) [inline src/main/java/com/exasol/exasoltestsetup/SessionBuilder.java:11]:
```suggestion
```

File: src/main/java/com/exasol/exasoltestsetup/SessionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    private String user;
+    private String host;
+    private int port;
+
+    // identity parameters
```
Comment (morazow) [inline src/main/java/com/exasol/exasoltestsetup/SessionBuilder.java:16]:
```suggestion
```

File: .settings/org.eclipse.core.resources.prefs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+eclipse.preferences.version=1
```
Comment (kaklakariada) [inline .settings/org.eclipse.core.resources.prefs:1]: delete and ignore this file

File: .settings/org.eclipse.m2e.core.prefs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+activeProfiles=
```
Comment (kaklakariada) [inline .settings/org.eclipse.m2e.core.prefs:1]: delete and ignore this file

File: .settings/org.eclipse.core.resources.prefs (hunk 3 above)
Comment (morazow) [inline .settings/org.eclipse.core.resources.prefs:1]: Are these required by new PK release?

**Review by morazow — DISMISSED** [review]

File: src/main/java/com/exasol/exasoltestsetup/testcontainers/ExasolTestcontainerTestSetup.java [hunk 5]
```diff
@@ -98,12 +98,15 @@ public void close() throws Exception {
...
-    private Session configSshAuth(final JSch ssh) throws JSchException {
-        final Session session = ssh.getSession("root", this.exasolContainer.getHost(),
-                this.exasolContainer.getMappedPort(SSH_PORT));
+    private SessionBuilder sessionBuilder() {
         final ByteArrayOutputStream privateKey = new ByteArrayOutputStream();
         this.keyPair.writePrivateKey(privateKey);
-        ssh.addIdentity("tmp-key", privateKey.toByteArray(), this.keyPair.getPublicKeyBlob(), null);
-        return session;
+        return new SessionBuilder() //
+                .identity("./cloudSetup/generated/exasol_cluster_ssh_key") //
```
Comment (kaklakariada) [inline src/main/java/com/exasol/exasoltestsetup/testcontainers/ExasolTestcontainerTestSetup.java:105]:
```suggestion
```


do we need this parameter? Who generates this file?

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — DISMISSED** [review]

File: src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java [hunk 6]
```diff
@@ -181,7 +182,10 @@ public SqlConnectionInfo getConnectionInfo() {
     public Bucket getDefaultBucket() {
         return SyncAwareBucket.builder().ipAddress("localhost").port(this.localBucketFsPort).name("default")
                 .serviceName("bfsdefault").readPassword(this.bucketFsReadPassword)
-                .writePassword(this.bucketFsWritePassword).monitor(new WaitBucketFsMonitor()).build();
+                .writePassword(this.bucketFsWritePassword) //
+                .monitor(new WaitBucketFsMonitor()) //
+                .stateRetriever(new TimestampRetriever()) //
```
Comment (kaklakariada) [inline src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java:187]: This line was missing and caused a nullpointer exception ;)

**Review by kaklakariada — DISMISSED** [review]

**Review by ckunki — COMMENTED** [review]
sdsd

**Review by kaklakariada — DISMISSED** [review]

File: src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java (hunk 6 above)
Comment (ckunki) [reply src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java:187]: the `.build()` was missing, right?

Comment (ckunki) [reply src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java:187]: OK, the stateRetriever! Damn!
Yes, 1000 thanks for the hint to add a check to the constructor!

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**